####Background

Epigenetic mechanisms, involving alterations to the genome that do not affect the underlying DNA sequence, are increasingly recognized as some of the principal mediators of gene expression (Duncan et al., 2014). The most researched and best understood epigenetic process is DNA methylation, which most commonly involves the addition of a methyl group to a cytosine in a CpG dinucleotide pair. The role of DNA methylation is best understood in mammals, where methylation in promoter regions has a repressive effect on gene expression (Jones and Takai, 2001). In plants and invertebrates, methylation of gene bodies prevails, and is thought to be the ancestral pattern (Zemach et al., 2010). Gene body methylation appears to have a range of functions, including regulating alternative splicing, repressing intragenic promoter activity, and reducing the efficiency of transcriptional elongation (Duncan et al., 2014). Methylation of gene bodies may vary according to gene function, and studies on invertebrates indicate that genes with housekeeping functions tend to be more heavily methylated than those with inducible functions (Roberts and Gavery, 2011, Sarda et al., 2012, Dixon et al., 2014, Gavery and Roberts, 2014). This has led to speculation that gene body methylation may promote predictable expression of essential genes for basic biological processes, while an absence of methylation could allow for stochastic transcriptional opportunities in genes involved in phenotypic plasticity (Roberts and Gavery 2012, Dixon et al., 2014, Gavery and Roberts, 2014).

Reef-building corals, the organisms that form the trophic and structural foundation of coral reef ecosystems, are among the numerous organisms about which we know very little regarding their epigenetics. However, it is known that corals display a significant degree of phenotypic plasticity (Todd 2008, Granados-Cifuentes et al., 2013). As long-lived, sessile organisms, corals are thought to be particularly reliant on phenotypic plasticity to cope with environmental heterogeneity, because they must be able to withstand whatever nature imposes on them over long periods of time (Bruno and Edmunds, 1997). Recent studies on gene expression variation support the view that phenotypic plasticity in corals is robust and may provide resilience in the face of ocean warming (Barshis et al., 2013, Granados-Cifuentes et al., 2013, Palumbi et al., 2014). However, the underlying basis of gene expression variation, and indeed phenotypic plasticity, remain largely unknown.

Evaluation of epigenetic processes therefore represents a logical next step in understanding coral gene expression and phenotypic variation. To date, only one study has investigated epigenetic processes in corals (Dixon et al., 2014). Germline DNA methylation patterns in the transcriptome of *Acropora millepora* corroborated findings reported in studies of other invertebrate species (Dixon et al., 2014). Most interestingly, genes that were differentially expressed in response to a common garden transplantation experiment were among the genes exhibiting lower levels of germline methylation (Dixon et al., 2014). This finding provides more direct support for earlier studies on invertebrates showing that hypomethylated genes tend to be those with inducible functions based on gene ontology (GO) information (Gavery and Roberts, 2010, Sarda et al., 2012).

In this study, patterns of germline methylation in several coral transcriptomes are assessed. Six coral species with published transcriptome data were chosen (Table 1). 

#####Table 1. Data sources used in the study
Organism | Transcriptome reference 
--------- | ---------- 
*Acropora hyacinthus* | Barshis et al. (2013) 
*Acropora millepora* | Moya et al. (2012) 
*Acropora palmata* | Polato et al. (2011) 
*Montastraea faveolata* | Desalvo et al. (2008) 
*Pocillopora damicornis* | Vidal-Dupiol et al. (2013) 
*Stylophora pistillata* | Karako-Lampert et al. (2014) 


Germline methylation patterns in these  data can be readily inferred due to the tendency of methylated cytosines to mutate into thymines over evolutionary time, which leads to reduced levels of CpG dinucleotides in heavily methylated regions. The ratio of observed to expected CpGs, known as CpG O/E or CpG ratio, can therefore be calculated to infer levels of germline methylation. Methylation levels estimated by this technique are highly correlated with direct measures of methylation enrichment (Sarda et al. 2012). In addition to calculating CpG ratio, transcriptome data were also annoted with GOSlim biological processes terms in order to relate patterns of germline methylation to information on gene function.

#####References

Barshis, D.J., Ladner, J.T., Oliver, T.A., Seneca, F.O., Traylor-Knowles, N., and Palumbi, S.R. (2013). Genomic basis for coral resilience to climate change. PNAS 110, 1387–1392.

Bruno, J.F., and Edmunds, P.J. (1997). Clonal variation for phenotypic plasticity in the coral Madracis mirabilis. Ecology 78, 2177–2190.

Desalvo, M.K., Voolstra, C.R., Sunagawa, S., Schwarz, J.A., Stillman, J.H., Coffroth, M.A., Szmant, A.M., and Medina, M. (2008). Differential gene expression during thermal stress and bleaching in the Caribbean coral Montastraea faveolata. Molecular Ecology 17, 3952–3971.

Dixon, G.B., Bay, L.K., and Matz, M.V. (2014). Bimodal signatures of germline methylation are linked with gene expression plasticity in the coral Acropora millepora. BMC Genomics 15, 1109.

Duncan, E.J., Gluckman, P.D., and Dearden, P.K. (2014). Epigenetics, plasticity, and evolution: How do we link epigenetic change to phenotype? J. Exp. Zool. B Mol. Dev. Evol. 322, 208–220.

Granados-Cifuentes, C., Bellantuono, A.J., Ridgway, T., Hoegh-Guldberg, O., and Rodriguez-Lanetty, M. (2013). High natural gene expression variation in the reef-building coral Acropora millepora: potential for acclimative and adaptive plasticity. BMC Genomics 14, 228.

Gavery, M.R., and Roberts, S.B. (2014). A context dependent role for DNA methylation in bivalves. Briefings in Functional Genomics elt054.

Gavery, M.R., and Roberts, S.B. (2010). DNA methylation patterns provide insight into epigenetic regulation in the Pacific oyster (Crassostrea gigas). BMC Genomics 11, 483.

Jones, P.A., and Takai, D. (2001). The role of DNA methylation in mammalian epigenetics. Science 293, 1068–1070.

Karako-Lampert, S., Zoccola, D., Salmon-Divon, M., Katzenellenbogen, M., Tambutté, S., Bertucci, A., Hoegh-Guldberg, O., Deleury, E., Allemand, D., and Levy, O. (2014). Transcriptome Analysis of the Scleractinian Coral Stylophora pistillata. PLoS ONE 9, e88615.

Maor-Landaw, K., Karako-Lampert, S., Ben-Asher, H.W., Goffredo, S., Falini, G., Dubinsky, Z., and Levy, O. (2014). Gene expression profiles during short-term heat stress in the red sea coral Stylophora pistillata. Glob Change Biol 20, 3026–3035.

Meyer, E., Aglyamova, G.V., and Matz, M.V. (2011). Profiling gene expression responses of coral larvae (Acropora millepora) to elevated temperature and settlement inducers using a novel RNA-Seq procedure. Molecular Ecology 20, 3599–3616.

Moya, A., Huisman, L., Ball, E.E., Hayward, D.C., Grasso, L.C., Chua, C.M., Woo, H.N., Gattuso, J.-P., Forêt, S., and Miller, D.J. (2012). Whole Transcriptome Analysis of the Coral Acropora millepora Reveals Complex Responses to CO2-driven Acidification during the Initiation of Calcification. Molecular Ecology 21, 2440–2454.

Polato, N.R., Altman, N.S., and Baums, I.B. (2013). Variation in the transcriptional response of threatened coral larvae to elevated temperatures. Mol Ecol 22, 1366–1382.

Polato, N.R., Vera, J.C., and Baums, I.B. (2011). Gene Discovery in the Threatened Elkhorn Coral: 454 Sequencing of the Acropora palmata Transcriptome. PLoS ONE 6, e28634.

Palumbi, S.R., Barshis, D.J., Traylor-Knowles, N., and Bay, R.A. (2014). Mechanisms of reef coral resistance to future climate change. Science 344, 895–898.

Roberts, S.B., and Gavery, M.R. (2011). Is there a relationship between DNA methylation and phenotypic plasticity in invertebrates? Frontiers in Physiology 2.

Sarda, S., Zeng, J., Hunt, B.G., and Yi, S.V. (2012). The Evolution of Invertebrate Gene Body Methylation. Mol Biol Evol 29, 1907–1916.

Shearer, T.L., Snell, T.W., and Hay, M.E. (2014). Gene Expression of Corals in Response to Macroalgal Competitors. PLoS ONE 9, e114525.

Todd, P.A. (2008). Morphological plasticity in scleractinian corals. Biological Reviews 83, 315–337.

Vidal-Dupiol, J., Dheilly, N.M., Rondon, R., Grunau, C., Cosseau, C., Smith, K.M., Freitag, M., Adjeroud, M., and Mitta, G. (2014). Thermal Stress Triggers Broad Pocillopora damicornis Transcriptomic Remodeling, while Vibrio coralliilyticus Infection Induces a More Targeted Immuno-Suppression Response. PLoS ONE 9, e107672.

Vidal-Dupiol, J., Zoccola, D., Tambutté, E., Grunau, C., Cosseau, C., Smith, K.M., Freitag, M., Dheilly, N.M., Allemand, D., and Tambutté, S. (2013). Genes Related to Ion-Transport and Energy Production Are Upregulated in Response to CO2-Driven pH Decrease in Corals: New Insights from Transcriptome Analysis. PLoS ONE 8, e58652.

Zemach, A., McDaniel, I.E., Silva, P., and Zilberman, D. (2010). Genome-wide evolutionary analysis of eukaryotic DNA methylation. Science 328, 916–919.


