The file blast2go_fasta_Pdamv2.fasta is a *Pocillopora damicornis* transcriptome assembly downloaded from http://2ei.univ-perp.fr/telechargement/transcriptomes/blast2go_fasta_Pdamv2.zip on 2/19/2015. It was originally featured in an article by [Vidal-Dupiol et al. (2013)](http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0058652). The transcriptome was developed from adult *P. damicornis* and consists of 72,890 contigs. According to the authors, "...27.7% and 69.8% of the contigs were predicted to belong to the symbiont and the host transcriptome, respectively. The remaining contigs could not be taxonomically attributed." 

The file OA_diff_contigs is a list of 48 differentially expressed contigs in response to ocean acidification (including putative dinoflagellate genes). It was derived from a [supplementary file](http://journals.plos.org/plosone/article/asset?unique&id=info:doi/10.1371/journal.pone.0058652.s001) from the same study noted above. Contig IDs are the same as those in the reference transcriptome and can be joined using common IDs. It is important to note, however, that this short list of contigs was chosen based on GO term enrichment analysis of nearly 9,000 contigs (16% of all contigs that were mapped to the reference transcriptome) that were differentially expressed in response to OA conditions. 

The file Immune_contigs is a list of 44 contigs involved in the innate immune response of *P. damicornis* to bacterial challenge. It was derived from a [supplementary file](http://journals.plos.org/plosone/article/asset?unique&id=info:doi/10.1371/journal.pone.0107672.s003) from an article by [Vidal-Dupiol et al. 2014](http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0107672). The study used the same reference transcriptome created in [Vidal-Dupiol et al. (2013)](http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0058652) and the contig IDs are the same, so the list of immune contigs can be joined with the transcriptome file using common IDs. As in the OA study, the 44 putative immune response genes were selected from a larger set of differentially expressed genes using a GO term enrichment analysis. 




